nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-api/src/main/java/org/apache/nifi/web/api/metrics/jmx/JmxMetricsCollector.java

```
@@ -0,0 +1,74 @@
+import java.util.Collection;
+import java.util.Set;
+
+public class JmxMetricsCollector {
+    private final static String PATTERN_FOR_ALL_OBJECT_NAMES = "*:*";
+    private final JmxMetricsResultConverter resultConverter;
+
+    public JmxMetricsCollector(final JmxMetricsResultConverter metricsResultConverter) {
+        this.resultConverter = metricsResultConverter;
+    }
+
+    public Collection<JmxMetricsResult> getBeanMetrics() {
+        final javax.management.MBeanServer mBeanServer = java.lang.management.ManagementFactory.getPlatformMBeanServer();
+        final Set<ObjectInstance> instances;
+        try {
+            instances = mBeanServer.queryMBeans(new ObjectName(PATTERN_FOR_ALL_OBJECT_NAMES), null);
+        } catch (MalformedObjectNameException e) {
+            throw new RuntimeException(e);
+        }
+
+        final Collection<JmxMetricsResult> results = new ArrayList<>();
+        for (final ObjectInstance instance : instances) {
+            final MBeanInfo info;
+            try {
+                info = mBeanServer.getMBeanInfo(instance.getObjectName());
+            } catch (InstanceNotFoundException | ReflectionException | IntrospectionException e) {
+                continue;
+            }
+
+            for (MBeanAttributeInfo attribute : info.getAttributes()) {
+                    try {
+                        final String beanName = instance.getObjectName().getCanonicalName();
+                        final String attributeName = attribute.getName();
+                        final Object attributeValue = resultConverter.convert(mBeanServer.getAttribute(instance.getObjectName(), attribute.getName()));
+
+                        results.add(new JmxMetricsResult(beanName, attributeName, attributeValue));
+                    } catch (MBeanException | RuntimeMBeanException | ReflectionException | InstanceNotFoundException | AttributeNotFoundException e) {
+                        //Empty or invalid attributes should not stop the loop.
```

Recommend introducing a trace log at this point for potential troubleshooting, instead of hiding the exception.